mlflow/utils/cli_args.py

```diff
@@ -45,12 +49,49 @@
 NO_CONDA = click.option(
     "--no-conda",
     is_flag=True,
-    help="If specified, will assume that MLmodel/MLproject is running within "
+    help="[Deprecated] If specified, will assume that MLmodel/MLproject is running within "
     "a Conda environment with the necessary dependencies for "
     "the current project instead of attempting to create a new "
     "conda environment.",
 )
 
+
+ENV_MANAGER = click.option(
+    "--env-manager",
+    default=None,
```

Comment:
Can we default to `conda` instead of defaulting to None and resolving later?

Reply:
If so, suppose old version users he might use command with argument `--no-conda`, then when upgrading to new version mlflow command will fail.

Reply:
currently `--no-conda` argument is deprecated not removed, so we should keep  `--no-conda`  works.

Reply:
@dbczumar I chose `None` to make it possible to detect whether or not `--env-manager` is specifed. We could 

```
contains_env_manager = "--env-manager" in sys.argv

if no_conda:
    # se
```

Reply:
Or we could do something like this:

```python
import sys


def _resolve_env_manager(ctx, param, value):
    no_conda = ctx.params["no_conda"]
    env_manager = EnvManager.from_string(value)
    if not no_conda:
        return env_manager

    is_env_manager_specified = param in sys.argv
    if not is_env_manager_specified:
        return EnvManager.LOCAL

    # Both --no-conda and --env-manager are specified. An exception should be thrown.
    raise Exception("...")


ENV_MANAGER = click.option(
    "--env-manager",
    default="conda",
    required=False,
    type=click.UNPROCESSED,
    callback=_resolve_env_manager,
    ...
)
```

Reply:
Or can we ignore `--env-manager` if `--no-conda` is supplied?

Reply:
Let's throw if both are supplied.

Reply:
We can leave this as `None` by default; that's fine :). I didn't see the issue before. Makes sense now!